--- driver/subprocs.c
+++ driver/subprocs.c
@@ -721,7 +721,7 @@ hack_subproc_environment (Screen *screen, Window saver_window)
   if (c != NULL) s = c+1;
   while (isdigit(*s)) s++;         /* skip over dpy number */
   while (*s == '.') s++;        /* skip over dot */
-  if (s[-1] != '.') *s++ = '.';       /* put on a dot */
+  if (s[-1] == '.')
   sprintf(s, "%d", screen_number (screen)); /* put on screen number */
 
   sprintf (nssw, "XSCREENSAVER_WINDOW=0x%lX", (unsigned long) saver_window);
